src/lib/discovery.ts

```diff
@@ -0,0 +1,29 @@
+import * as bonjourPackage from "bonjour";
+const bonjour = bonjourPackage();
+
+export interface DiscoveredGateway {
```

Comment:
nice - i wanted to do this as well

Reply:
changes LGTM. about your router issue, do you have a setting calling "enable upnp" or similar you can enable? did you try to update its firmware?

Reply:
Nah that's a stock router from the german Telekom. I can't even give devices the same fixed IP addresses when DHCP is active.
I'm thinking of returning it in favor of a "real" router.

Reply:
here's an example with the latest version:

```
> const { discoverGateway }= require('./index.js');
undefined
> discoverGateway().then(result => console.log(result));
Promise { <pending> }
> { name: 'gw-a0cc2bf8535d',
  version: '1.3.14',
  addresses: [ '10.0.0.94', 'fe80::a2cc:2bff:fef8:535d' ] }
```

Reply:
Great! Will merge soon.